sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -242,22 +242,21 @@ object DateTimeUtils {
             return None
           }
         } else if (i == 5 || i == 6) {
-          if (b == 'Z') {
+          if (b == '-' || b == '+') {
```

Comment:
`getZoneId()` is able to handle zone offsets w/ prefix `-` and `+` but it doesn't support the format `7:3` like in https://github.com/apache/spark/blob/ffc0935e64ddd56c40db1c1c7f3afb7e5e306861/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala#L164 . So, I have to keep the code for backward compatibility.